### ISSUE TITLE
Add support for Uniswap

### ISSUE THREAD

**Issue by grahamperich** [grahamperich/ethereum-dex-prices-service#4]
This one looks like it will be a bit more challenging since there is no HTTP API available. The docs are here for anyone who wants to give it a shot: https://docs.uniswap.io/

@conlan on Twitter is working on an HTTP API that will make it much easier.

Until then you can pull all the NewExchange events from the factory: 
0xc0a47dFe034B400B47bDaD5FecDa2621de6c4d95

Or use getExchange() functions on the factory:
https://docs.uniswap.io/frontend-integration/connect-to-uniswap

or

https://github.com/Uniswap/uniswap-info/blob/master/src/constants/Uniswap.js#L4

**Comment by haydenadams** [comment]
If you guys are open to using web3 directly, prices are very easy. There is a mapping of `tokenAddress -> exchangeAddress` which you can get from the now finished directory API endpoint, or can just pull all NewExchange events. Or just call `getExchange(tokenAddress)` on the factory.

If I want to calculate the cost of buying/selling DAI, all you need is the ETH and DAI balance of the DAI exchange contract:

```javascript
ethReserve = web3.eth.getBalance(daiExchangeAddress)
erc20Reserve = daiContract.balanceOf(daiExchangeAddress)
```

then you just plug into either the buy or sell formula, as implemented in the contract:

```javascript
/*
If you're selling ERC20 for ETH, sold reserve is erc20Reserve, boughtReserve is ethReserve
If you're selling ETH for ERC20, sold reserve is ethReserve, boughtReserve is erc20Reserve
*/
function sellPrice(amountSold, soldReserve, boughtReserve) {
   numerator = amountSold * boughtReserve * 997
   denominator = soldReserve * 1000 + amountSold * 997
   amountBought = numerator / denominator
}

/*
If you're buying ERC20 with ETH, sold reserve is ethReserve, boughtReserve is erc20Reserve
If you're buying ETH with ERC20, sold reserve is erc20Reserve, boughtReserve is ethReserve
*/
function buyPrice(amountBought, soldReserve, boughtReserve) {
   numerator = amountBought * soldReserve * 1000
   denominator = (boughtReserve - amountBought) * 997
   amountSold = numerator / denominator + 1
}
```

**Comment by grahamperich** [comment]
Thanks @haydenadams ! Seems straight forward from here.

**Comment by grahamperich** [comment]
resolved by #12
